test/sql/function/string/regex_extract.test

```diff
@@ -0,0 +1,97 @@
+# name: test/sql/function/string/regex_extract.test
+# description: regex extract test
+# group: [string]
+
+statement ok
+PRAGMA enable_verification
+
+query T
+SELECT regexp_extract('foobarbaz', 'b..')
+----
+bar
+
+query T
+SELECT regexp_extract('foobarbaz', 'b..', 1)
+----
+(empty)
+
+query T
+SELECT regexp_extract('foobarbaz', '(b..)(b..)')
+----
+barbaz
+
+query T
+SELECT regexp_extract('foobarbaz', '(b..)(b..)', 1)
+----
+bar
+
+query T
+SELECT regexp_extract('foobarbaz', '(b..)(b..)', 2)
+----
+baz
+
+statement ok
+CREATE TABLE test (s VARCHAR, p VARCHAR, i INT)
+
+statement ok
+INSERT INTO test VALUES
+    ('foobarbaz', 'b..', 0),
+    ('foobarbaz', 'b..', 1),
+    ('foobarbaz', '(b..)(b..)', 0),
+    ('foobarbaz', '(b..)(b..)', 1),
+    ('foobarbaz', '(b..)(b..)', 2)
+
+query T
+SELECT regexp_extract(s, p, i) FROM test
+----
+bar
+(empty)
+barbaz
+bar
+baz
+
+query T
+SELECT regexp_extract(s, p, 0) FROM test
+----
+bar
+bar
+barbaz
+barbaz
+barbaz
+
+query T
+SELECT regexp_extract(s, 'b..', 0) FROM test
+----
+bar
+bar
+bar
+bar
+bar
+
+query T
+SELECT regexp_extract(s, '(b..)(b..)', i) FROM test
+----
+barbaz
+bar
+barbaz
+bar
+baz
+
+# null values
+query T
+SELECT regexp_extract('foobarbaz', NULL, 0)
+----
+(empty)
+
+query T
+SELECT regexp_extract('foobarbaz', 'b..', NULL)
```

Comment:
Could you add a test case where the group index is out of range or negative?

Reply:
So this was interesting. I tried a group index > 9 and RE2 would not allow it. So, for now I've had to require the group_idx to be constant, and between 0 and 9.

Reply:
Hmm, would it be possible to return a list of all the groups and just do a list slice afterwards (instead of passing in the parameter)? It would be slower, but you could look at a higher group number.

Reply:
RE2's extract requires 4 parameters. The input string, the regex pattern, the 'rewrite' field, and a pointer where it can write the output string.

The problem here is the 'rewrite' field. This allows selecting the capture group with this syntax: `'\\1'` selects the first capture group, `'\\2'` selects the second capture group, etc.

As such, it is impossible to select the 42nd capture group, as we cannot escape two characters e.g. `'\\42'`.

Of course, it is possible to implement, but it would be cumbersome, as we would have to hand-roll the function, rather than using RE2 directly.

Also, I doubt that people want the 42nd capture group anyway. They can change their pattern to get what they want.